HLTrigger/Muon/plugins/HLTMuonL3orL2PreFilter.cc

```diff
@@ -0,0 +1,376 @@
+/** \class HLTMuonL3orL2PreFilter
+ *
+ *
+ *  This class is an HLTFilter (-> EDFilter) implementing
+ *  a simple filter to select L3 muons and L2 if they dont match with any L3
+ * 
+ *  Original author:  A. Soto <alejandro.soto.rodriguez@cern.ch>
+ *
+ */
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/HLTReco/interface/TriggerFilterObjectWithRefs.h"
+#include "DataFormats/HLTReco/interface/TriggerRefsCollections.h"
+#include "DataFormats/RecoCandidate/interface/RecoChargedCandidate.h"
+#include "DataFormats/RecoCandidate/interface/RecoChargedCandidateFwd.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "HLTrigger/HLTcore/interface/HLTFilter.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+class HLTMuonL3orL2PreFilter : public HLTFilter {
+public:
+  explicit HLTMuonL3orL2PreFilter(const edm::ParameterSet&);
+  ~HLTMuonL3orL2PreFilter() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  bool hltFilter(edm::Event&,
+                 const edm::EventSetup&,
+                 trigger::TriggerFilterObjectWithRefs& filterproduct) const override;
+
+  bool checkOverlap(const reco::RecoChargedCandidate r1,
+                    const reco::RecoChargedCandidate r2,
+                    const double maxDeltaR2) const {
+    return (deltaR2(r1, r2) < maxDeltaR2);
+  }
+
+  double pTeffTransformation(reco::RecoChargedCandidateRef candidate, reco::TrackRef track) const {
+    auto const pt = candidate->pt();
+    auto const err0 = track->error(0);
+    auto const abspar0 = std::abs(track->parameter(0));
+    // if abspar0 > 0, convert 50% efficiency threshold to 90% efficiency threshold
+    auto const ptLx = abspar0 > 0 ? pt + nsigma_Pt_ * err0 / abspar0 * pt : pt;
+    return ptLx;
+  }
```

Comment:
```suggestion
  double pTeffTransformation(reco::RecoChargedCandidateRef const& candidate, reco::TrackRef const& track) const;
```
Please move implementation outside of class definition (see comment below).